ansible_collections/arista/avd/roles/eos_designs/docs/how-to/custom-templates.md

```diff
@@ -0,0 +1,67 @@
+<!--
+  ~ Copyright (c) 2023 Arista Networks, Inc.
+  ~ Use of this source code is governed by the Apache License 2.0
+  ~ that can be found in the LICENSE file.
+  -->
+
+# Custom Templates
+
+With Custom Templates for the `eos_designs` role, the user can create additional structured configuration variables that would otherwise not be created by the builtin templates.
+These additional variables will then be available for use by other AVD roles.
+
+For example, the user may require additional structured configuration variables to render a custom template the user has created for the `eos-cli-config-gen` role.
+
+## Create a Custom Template
+
+Custom templates should be written in Jinja2.
+
+For details on how to create Jinja2 templates, see [Templating Jinja2](https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_templating.html#templating-jinja2)
+
+For example:
+```jinja2
+_structured_configuration_variable_name: structured_configuration_variable_value {{ some_variable }}
+```
+
+At runtime, `eos_designs` searches for custom templates in several locations, therefore, custom templates should be saved in one of these locations:
+
+```commandline
+~/.ansible/collections/ansible_collections/arista/avd/roles/eos_designs/templates/templates/<template name>
+~/.ansible/collections/ansible_collections/arista/avd/roles/eos_designs/templates/<template name>
+~/.ansible/collections/ansible_collections/arista/avd/roles/eos_designs/<template name>
+~/.ansible/collections/ansible_collections/arista/avd/roles/eos_designs/tasks/templates/templates/<template name>
+~/.ansible/collections/ansible_collections/arista/avd/roles/eos_designs/tasks/templates/<template name>
+~/.ansible/collections/ansible_collections/arista/avd/roles/eos_designs/tasks/<template name>
```

Comment:
We should avoid suggesting to update the collection files, since it would be lost on an upgrade.